src/usb_host.c

```diff
@@ -132,19 +131,17 @@ static void USBH_UserProcess(USBH_HandleTypeDef *phost, uint8_t id) {
 
     case HOST_USER_DISCONNECTION:
         Appli_state = APPLICATION_DISCONNECT;
-        media_set_inserted(false);
+        media_set_removed();
         f_mount(0, (TCHAR const *)USBHPath, 1); //umount
         break;
 
     case HOST_USER_CLASS_ACTIVE:
         Appli_state = APPLICATION_READY;
         FRESULT result = f_mount(&USBHFatFS, (TCHAR const *)USBHPath, 0);
         if (result == FR_OK)
-            media_set_inserted(true);
-        else {
-            // TODO: inform the user
-            _dbg("disk not mounted; error: %d", result);
-        }
+            media_set_inserted();
+        else
+            media_set_error(media_error_MOUNT);
```

Comment:
How do we visualize this for the user? Do we have an icon for this state, so that our tech-support guys can immediately tell the user, that his USB flash wasn't recognized?

Reply:
We can improve this - now there is 3 state logic - inserted/removed/error. Maybe some next pull :)